classes/event/cert_regenerated.php

```diff
@@ -0,0 +1,58 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * cert_regenerated class.
+ *
+ * @package    auth_saml2
+ * @author     Kristian Ringer <kristianringer@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace auth_saml2\event;
+
+use core\event\base;
+use moodle_url;
+
+defined('MOODLE_INTERNAL') || die();
+
+/**
+ * cert_regenerated class.
+ *
+ * @package    auth_saml2
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class cert_regenerated extends base {
+    /**
+     * Returns non-localised event description with id's for admin use only.
+     *
+     * @return string
+     */
+    public function get_description() {
+        $byuser = '';
+        $this->other['userid'] ? $byuser .= "by user '{$this->other['userid']}'" : null;
```

Comment:
I think a simple if would be more readable here. also '' could default to 'unknown'

Reply:
Have updated it to read more clearly now